### DISCUSSION TITLE
How can i update an App on Linux ?

### DISCUSSION THREAD

**Discussion by pkarsten** [unoplatform/uno#18564] [General]
I have my UNO .NET app on a Raspberry Pi with Linux. Now I would like to install an update routine. What is the best way to do this? At the moment I copy the files to the Linux and start the APP with “dotnet MyApp.dll” . When I deploy the files (I develop with Visual Studio), and then make them available in the cloud on a shared directory. How can I replace them on the Linux? Can I do this from within my app? I'm imagining a button that downloads and replaces the files, but that probably won't work because the DLL file is currently in use. How could I manage an update? Currently the app works on Linux, Windows and Android, would of course be nice if I could implement the routine for all 3. But the Raspberry is my primary target. Does anyone have any ideas or approaches for me? 

Thank you very much 

Peter

**Comment by jeromelaban** [comment]
Thanks for the question. So far, we do not provide any auto update solution (similar to click once, or like a store deployment would) for desktop. Snap could be a solution, but it's probably too heavy for a RPi.

Still, you can implement something similar by having the app download an app manifest (to check for version and authenticity) and a zip file, then to run an external script that would in-place update the app then restart it.

**Reply by pkarsten** [reply]
Thank you for your answer. I think I will write an update APP. The main app then checks a file in the cloud for version, and when update is available, you want to close the main app and run the update app, which picks up the files and copies them to the appropriate directory. But first look, because the app, including Raspberry, is currently only distributed to selected people in the family, but I think that could be a solution.
